Source/WebKit/UIProcess/Extensions/WebExtension.cpp

```diff
@@ -1137,6 +1147,229 @@ void WebExtension::populateContentScriptPropertiesIfNeeded()
+    return m_externallyConnectableMatchPatterns;
+}
+
+WebExtension::MatchPatternSet WebExtension::allRequestedMatchPatterns()
+{
+    populatePermissionsPropertiesIfNeeded();
+    populateContentScriptPropertiesIfNeeded();
+    populateExternallyConnectableIfNeeded();
+
+    WebExtension::MatchPatternSet result;
+
+    for (auto& matchPattern : m_permissionMatchPatterns)
+        result.add(matchPattern);
+
+    for (auto& matchPattern : m_externallyConnectableMatchPatterns)
+        result.add(matchPattern);
+
+    for (auto& injectedContent : m_staticInjectedContents) {
+        for (auto& matchPattern : injectedContent.includeMatchPatterns)
+            result.add(matchPattern);
+    }
+
+    return result;
+}
+
+void WebExtension::populateExternallyConnectableIfNeeded()
+{
+    if (m_parsedExternallyConnectable)
+        return;
+
+    m_parsedExternallyConnectable = true;
+
+    RefPtr manifestObject = this->manifestObject();
+    if (!manifestObject)
+        return;
+
+    // Documentation: https://developer.mozilla.org/docs/Mozilla/Add-ons/WebExtensions/manifest.json/externally_connectable
+
+    RefPtr externallyConnectableObject = manifestObject->getObject(externallyConnectableManifestKey);
+    if (!externallyConnectableObject)
+        return;
+
+    if (!externallyConnectableObject->size()) {
+        recordError(createError(Error::InvalidExternallyConnectable));
+        return;
+    }
+
+    bool shouldReportError = false;
+    MatchPatternSet matchPatterns;
+
+    if (RefPtr matchPatternStrings = externallyConnectableObject->getArray(externallyConnectableMatchesManifestKey)) {
+        for (auto matchPatternStringValue : *matchPatternStrings) {
+            auto matchPatternString = matchPatternStringValue->asString();
+
+            if (matchPatternString.isEmpty())
+                continue;
+
+            if (auto matchPattern = WebExtensionMatchPattern::getOrCreate(matchPatternString)) {
+                if (matchPattern->matchesAllURLs() || !matchPattern->isSupported()) {
+                    shouldReportError = true;
+                    continue;
+                }
+
+                // URL patterns must contain at least a second-level domain. Top level domains and wildcards are not standalone patterns.
+                if (matchPattern->hostIsPublicSuffix()) {
+                    shouldReportError = true;
+                    continue;
+                }
+
+                matchPatterns.add(matchPattern.releaseNonNull());
+            }
+        }
+    }
+
+    m_externallyConnectableMatchPatterns = matchPatterns;
+
+    RefPtr extensionIDs = externallyConnectableObject->getArray(externallyConnectableIDsManifestKey);
+    if (extensionIDs) {
+        extensionIDs = filterObjects(*extensionIDs, [](auto& value) {
+            return !value.asString().isEmpty();
+        });
+    }
+
+    if (shouldReportError || (matchPatterns.isEmpty() && (!extensionIDs || (extensionIDs && !extensionIDs->length()))))
```

Comment:
Since the `WTFMove()` I added above, need to use `m_externallyConnectableMatchPatterns` here.
```suggestion
    if (shouldReportError || (m_externallyConnectableMatchPatterns.isEmpty() && (!extensionIDs || (extensionIDs && !extensionIDs->length()))))
```